From 705050f46e477677a5cd604244a6b3b711344cfe Mon Sep 17 00:00:00 2001
From: Bastian Bittorf <bittorf@bluebottle.com>
Date: Thu, 4 Jun 2015 12:36:59 +0200
Subject: [PATCH] [base-files] allow telnet login without password

allow login without password, when dropbear is missing.
ask for credentials, when /etc/httpd.conf is available.

Signed-off-by: Bastian Bittorf <bittorf@bluebottle.com>
---
 package/base-files/files/bin/login.sh |   20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/package/base-files/files/bin/login.sh b/package/base-files/files/bin/login.sh
index 25627b6..67c2eb3 100755
--- a/package/base-files/files/bin/login.sh
+++ b/package/base-files/files/bin/login.sh
@@ -5,8 +5,24 @@ if ( ! grep -qsE '^root:[!x]?:' /etc/shadow || \
      ! grep -qsE '^root:[!x]?:' /etc/passwd ) && \
    [ -z "$FAILSAFE" ]
 then
-	echo "Login failed."
-	exit 0
+	if [ -x /usr/sbin/dropbear ]; then
+		echo "Login failed."
+		exit 0
+	else
+		if [ -e '/etc/httpd.conf' ]; then
+			echo 'enter root-password: '
+			read -r PASS
+
+			if [ "$PASS" = "$( cut -d: -f3 /etc/httpd.conf )" ]; then
+				echo 'OK'
+			else
+				echo 'password wrong'
+				exit 0
+			fi
+		else
+			echo 'OK: login without password'
+		fi
+	fi
 else
 cat << EOF
  === IMPORTANT ============================
-- 
1.7.10.4

